### PR TITLE
Fixes post 3.2.0 release

### PR DESCRIPTION
**Pull request by arthur740212**
The input login was inverted during the refactor and was causing breakages when interacting with the overlay.
This fix applies the working logic.

### CODE REVIEW

File: Assets/Scripts/UI/Login/UILoginMenu.cs [hunk 1]
```diff
@@ -319,29 +319,37 @@ private static void PreventDeselection(ref GameObject previouslySelectedGameObje
...
-            return true;
+#if ENABLE_DEBUG_INPUT
+            LogInputChanged(shouldHandle);
 #endif
+            return shouldHandle;
+
         }
 
+#if ENABLE_DEBUG_INPUT
+        static bool previousShouldHandle = false;
```
Comment (paulhazen) [inline Assets/Scripts/UI/Login/UILoginMenu.cs]: I believe our practice for this circumstance would be for this to be named `s_previousShouldHandle`. Either way, I think it's probably appropriate for it to be made `private`.

Separately, is it guaranteed that the previous state of whether input should be handled is `false`?

File: Assets/Scripts/UI/Login/UILoginMenu.cs [hunk 2]
```diff
@@ -319,29 +319,37 @@ private static void PreventDeselection(ref GameObject previouslySelectedGameObje
...
+        static void LogInputChanged(bool shouldHandle)
+        {
+            bool result = previousShouldHandle != shouldHandle ? true : false;
+            if (result)
```
Comment (paulhazen) [inline Assets/Scripts/UI/Login/UILoginMenu.cs]: I think it would be clearer if you changed this to be

```c#
if (previousShouldHandle != shouldHandle)
```
(including suggested changes from earlier comment)

That way you can entirely remove line `345`.

File: Assets/Scripts/UI/Login/UILoginMenu.cs [hunk 3]
```diff
@@ -319,29 +319,37 @@ private static void PreventDeselection(ref GameObject previouslySelectedGameObje
...
+#if ENABLE_DEBUG_INPUT
+            LogInputChanged(shouldHandle);
 #endif
+            return shouldHandle;
+
         }
 
+#if ENABLE_DEBUG_INPUT
+        static bool previousShouldHandle = false;
+        static void LogInputChanged(bool shouldHandle)
```
Comment (paulhazen) [inline Assets/Scripts/UI/Login/UILoginMenu.cs]: As far as I can tell, I think this should be marked as `private`.

File: Assets/Scripts/UI/Login/UILoginMenu.cs [hunk 4]
```diff
@@ -319,29 +319,37 @@ private static void PreventDeselection(ref GameObject previouslySelectedGameObje
...
+        {
+            bool result = previousShouldHandle != shouldHandle ? true : false;
+            if (result)
+            {
+                Debug.LogWarning($"Input {(shouldHandle ? "enabled" : "disabled")} for main app");
+            }
+            previousShouldHandle = shouldHandle;
```
Comment (paulhazen) [inline Assets/Scripts/UI/Login/UILoginMenu.cs]: I believe that this assignment should be within the conditional.

File: Assets/Scripts/UI/Login/UILoginMenu.cs [hunk 5]
```diff
@@ -319,29 +319,37 @@ private static void PreventDeselection(ref GameObject previouslySelectedGameObje
...
         }
 
+#if ENABLE_DEBUG_INPUT
+        static bool previousShouldHandle = false;
+        static void LogInputChanged(bool shouldHandle)
+        {
+            bool result = previousShouldHandle != shouldHandle ? true : false;
+            if (result)
+            {
+                Debug.LogWarning($"Input {(shouldHandle ? "enabled" : "disabled")} for main app");
```
Comment (paulhazen) [inline Assets/Scripts/UI/Login/UILoginMenu.cs:355]: I've found that for clarity it can be useful to (instead of having an inline ternary statement) rephrase the log so it makes sense for the value `true` or `false` to be used. Something like:
```C#
Debug.LogWarning($"Should input be handled: {shouldHandle}.");
```

This is _not_ a critique relevant exactly for this PR - since you merely relocated the log line - and (if memory serves) I'm the one who originally wrote it 😆

File: Assets/Scripts/UI/Login/UILoginMenu.cs [hunk 6]
```diff
@@ -319,29 +319,37 @@ private static void PreventDeselection(ref GameObject previouslySelectedGameObje
         /// <returns>True if input should be handled, false if not.</returns>
         private static bool ShouldInputBeHandled()
         {
-            // TODO: Clarify why this is only evaluated for PS4 and PS5
-#if UNITY_PS4 || UNITY_PS5
-            // TODO: Simplify this conditional - it's sort of confusing to read despite doing exactly what we want
-            if (null == EventSystem.current || EventSystem.current.sendNavigationEvents == !EOSManager.Instance.IsOverlayOpenWithExclusiveInput())
```
Comment (paulhazen) [inline Assets/Scripts/UI/Login/UILoginMenu.cs:325]: I think we still need to check if `EventSystem.current` is `null`. While it was far from perfect, this conditional _did_ provide safety against referencing a null pointer.

**Review by paulhazen — CHANGES_REQUESTED** [review]

File: .gitattributes [hunk 7]
```diff
@@ -30,6 +30,4 @@
 *.physicMaterial    merge=unityyamlmerge eol=lf
 *.asset             merge=unityyamlmerge eol=lf
 *.meta              merge=unityyamlmerge eol=lf
-*.controller        merge=unityyamlmerge eol=lf
-
-Assets/Plugins/Linux/libDynamicLibraryLoaderHelper.so filter=lfs diff=lfs merge=lfs binary
```
Comment (WispyMouse) [inline .gitattributes:35]: Is this removal intentional?

File: Assets/Scripts/UI/Login/UILoginMenu.cs (hunk 6 above)
Comment (WispyMouse) [reply Assets/Scripts/UI/Login/UILoginMenu.cs:325]: Agreeable; it seems like we're using EventSystem.current without checking that it exists, later in this function.

File: Assets/Scripts/UI/Login/UILoginMenu.cs [hunk 8]
```diff
@@ -319,29 +319,37 @@ private static void PreventDeselection(ref GameObject previouslySelectedGameObje
...
             bool shouldHandle = !EOSManager.Instance.IsOverlayOpenWithExclusiveInput();
-                
-            Debug.Log($"Input {(shouldHandle ? "enabled" : "disabled")} due to EOS Overlay.");
+
 #if ENABLE_INPUT_SYSTEM
             EventSystem.current.currentInputModule.enabled = shouldHandle;
+            EventSystem.current.sendNavigationEvents = shouldHandle;
 #else
+            // TODO: Clarify why this is only evaluated for PS4 and PS5
+#if UNITY_PS4 || UNITY_PS5
```
Comment (WispyMouse) [inline Assets/Scripts/UI/Login/UILoginMenu.cs:337]: Is this an appropriate place for UNITY_PS4 and UNITY_PS5 specific code? Would this be appropriate to always set like this, even if ENABLE_INPUT_SYSTEM is not true?

File: Assets/Scripts/UI/Login/UILoginMenu.cs (hunk 1 above)
Comment (arthur740212) [reply Assets/Scripts/UI/Login/UILoginMenu.cs]: It is the input state during startup, so it does not matter whether it is false or true.
It is private already, but is has to be static if used in a static function, I could add a private keyword to be precise

**Review by WispyMouse — CHANGES_REQUESTED** [review]
I think it is probably necessary to nullcheck against EventSystem.current, like it used to. I can be convinced otherwise.

File: Assets/Scripts/UI/Login/UILoginMenu.cs (hunk 2 above)
Comment (arthur740212) [reply Assets/Scripts/UI/Login/UILoginMenu.cs]: yep, there used to be a `return result;`, then the function was rewritten into a void, so it is not needed anymore, nice find.

File: .gitattributes (hunk 7 above)
Comment (arthur740212) [reply .gitattributes:35]: Sort of, this PR was supposed to be all fixes post 3.2.0, there was only one thus the name.
I'll rename the PR to reflect the branch

File: Assets/Scripts/UI/Login/UILoginMenu.cs (hunk 3 above)
Comment (arthur740212) [reply Assets/Scripts/UI/Login/UILoginMenu.cs]: It is private, will add the keyword for less confusion.

File: Assets/Scripts/UI/Login/UILoginMenu.cs (hunk 4 above)
Comment (arthur740212) [reply Assets/Scripts/UI/Login/UILoginMenu.cs]: Yep, putting it out doesn't do more because we only need to reassign if the values changed, nice catch.

File: Assets/Scripts/UI/Login/UILoginMenu.cs (hunk 6 above)
Comment (arthur740212) [reply Assets/Scripts/UI/Login/UILoginMenu.cs:325]: Fixed

File: .gitattributes (hunk 7 above)
Comment (andrew-hirata-playeveryware) [reply .gitattributes:35]: Both those removes seems suspicious to me.

Comment (arthur740212) [reply .gitattributes:35]: Only the linux line is actually removed, the other one is because of line ending changes

**Review by andrew-hirata-playeveryware — APPROVED** [review]

File: .gitattributes (hunk 7 above)
Comment (arthur740212) [reply .gitattributes:35]: The linux line is moved to the `.gitattribute` under `Linux/`
Since there is no going back from lfs I suppose

**Review by WispyMouse — APPROVED** [review]

**Review by paulhazen — APPROVED** [review]
